examples/sink/fallback/index.js

```diff
@@ -6,7 +6,7 @@ const fbSink = {
         for (const datum of datumStream) {
             const response = {
                 id: datum.id,
-                status: 'FALLBACK',
+                fallback: true,
```

Comment:
what will happen if i also set `success` has `true` along with `fallback`.
```suggestion
                fallback: true,
                success: true
```

Reply:
the behaviour would be same as we check for `fallback` in the response first ([at least in GO sdk](https://github.com/numaproj/numaflow-go/blob/47460a1854b8f58a0e918056ef4d169949193ebe/pkg/sinker/service.go#L215) )

Similarly [we implemented in this sdk](https://github.com/numaproj/numaflow-js/blob/37f1e5b2be1be449b592fe2abecfe32856997a44/src/sink/service.ts#L133) as well.

Once the status is set to fallback, on the numa side we will[ treat it as fallback message](https://github.com/numaproj/numaflow/blob/76ecdc63b439e24b9acd025dfba5d47c47fed25b/rust/numaflow-core/src/sink.rs#L713).

Interesting thing I observed now is that if `success` is true and `serve` is also true then we may end up not treating that message for serving ([as we match success first at numa](https://github.com/numaproj/numaflow/blob/76ecdc63b439e24b9acd025dfba5d47c47fed25b/rust/numaflow-core/src/sink.rs#L708))

Reply:
#### Tested with both fallback and success as true

Result Map that numa receives will have `Fallback` as status

```
2025-05-07T23:12:45.282852Z  INFO numaflow_core::sink: Starting sink writer self.batch_size=500 self.chunk_timeout=1s
2025-05-07T23:12:46.291003Z  INFO numaflow_core::sink: result map: {"simple-mono-vertex-1746659565283030880-0-0": Fallback, "simple-mono-vertex-1746659565283030171-0-0": Fallback, "simple-mono-vertex-1746659565283031088-0-0": Fallback, "simple-mono-vertex-1746659565282974130-0-0": Fallback, "simple-mono-vertex-1746659565283030630-0-0": Fallback}
2025-05-07T23:12:46.291021Z  INFO numaflow_core::sink: reached fallback loop
```

Reply:
@yhl25 i think we need to revisit this. the current implementation (even in golang) is ambiguous.